explore/src/main/scala/explore/targeteditor/AsterismEditor.scala

```diff
@@ -79,23 +80,24 @@ object AsterismEditor {
   protected val component =
     ScalaFnComponent
       .withHooks[Props]
-      // selectedTargetIdState
-      .useStateBy(_.asterism.get.headOption.map(_.id))
       // adding
       .useState(false)
       // reset "loading" for add button when science targets change, which indicates server roundtrip is over
-      .useEffectWithDepsBy((props, _, _) => props.asterism.get)((_, _, adding) =>
+      .useEffectWithDepsBy((props, _) => props.asterism.get)((_, adding) =>
         _ => adding.setState(false)
       )
-      .renderWithReuse { (props, selectedTargetIdState, adding) =>
+      .useEffectWithDepsBy((props, _) => (props.asterism.get, props.selectedTargetId.get))(
```

Comment:
We could potentially just use `useEffectOnMountBy` to set the selected target to the first one in the list when the component is mounted. Doing it with `useEffectWithDepsBy` has the effect of trying to keep whatever target was last selected to still be selected. This happens even if that target was in a different asterism, which I haven't decided if I like. On the other hand, with a little more expansion of the code, we could select a particular target in an asterism from the target summary table - which I think I do like.